File: server/src/main/java/com/example/server/entity/Room.java
package com.example.server.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.List;

@Entity
@Table(name = "rooms")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Room {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private int id;

    @Column(name = "name")
    private String name;

    @Column(name = "description")
    private String description;

    @Column(name = "status")
    private boolean status;

    @Column(name = "price")
    private float price;

    @Column(name = "max_user")
    private  int max_user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "hotel_id")
    @JsonIgnore
    private Hotel hotel;

    @OneToMany(cascade = CascadeType.ALL, mappedBy = "room", fetch = FetchType.LAZY)
    @JsonIgnore
    private List<User> users;

    @ManyToMany
    @JsonIgnore
    @JoinTable(
            name = "room_device",
            joinColumns = @JoinColumn(name = "room_id"),
            inverseJoinColumns = @JoinColumn(name = "device_id")
    )
    private List<Device> devices;
}

File: hotel-managment/src/apis/room.ts
import { Room } from "../types/room";
import { callApi } from "./callAPI";

export const createRoom = async (
    data: {
        name?: string,
        description?: string,
        status?: boolean | string | null,
        price?: number,
        max_user?: number,
        hotel_id?: number
    }
) => {
    return await callApi<Room>("room/create-room", 'post', data)
}

export const getAllRoom = async () => {
    return await callApi<Room[]>("room/get-all", 'get')
}

export const getRoomByHotelId = async (
    data: {
        hotel_id: number
    }
) => {
    return await callApi<Room[]>("room/get-all", 'get', data)
}

export const searchRoom = async (
    data: {
        name?: string 
        category?: string
    }
) => {
    return await callApi<Room[]>("room/search-room", 'post', data)
}